### DISCUSSION TITLE
Tweak style for PasswordRevealer Text

### DISCUSSION THREAD

**Discussion by jochenberger** [EclipseNebula/nebula#456] [Q&A]
Hi,

I'd like to use a `PasswordRevealer` next to a `Text` in my app. The styling is somehow off unfortunately:
![grafik](https://user-images.githubusercontent.com/370733/161923098-118906b6-b6a0-46a2-bdbb-5fcfc71598b8.png)
My "User" `Text` has `SWT.BORDER` set and I'm using Gnome, that's probably why I get those rounded corners. But I cannot style the embedded `Text` of the `PasswordRevealer`.
Any ideas?

Thanks,
Jochen

**Comment by wimjongman** [comment]
Can you take a look at the source code and come up with a suggestion and a PR? It should not be hard at all because the PwR uses a Text as an underlying widget.

**Reply by wimjongman** [reply]
https://github.com/eclipse/nebula/blob/master/widgets/passwordrevealer/org.eclipse.nebula.widgets.passwordrevealer/src/org/eclipse/nebula/widgets/passwordrevealer/PasswordRevealer.java

**Reply by jochenberger** [reply]
I'm not sure if it's even possible. The border is applied to a composite that wraps the Text and the Button. I don't think that we want an additional border around the Text. I'm not sure how to fix this.

**Reply by wimjongman** [reply]
I think the border should not be applied to the composite but to the wrapped text.

**Reply by wimjongman** [reply]
line 130 already processes the passed style:

       passwordField = new Text(this, removeFields(style, SWT.BORDER, SWT.PUSH) | SWT.PASSWORD);

**Reply by wimjongman** [reply]
So maybe the composite should be SWT.NONE and the SWT.BORDER should not be removed from the style.

**Reply by jochenberger** [reply]
But wouldn't the button be outside of the border then? I'm not sure if that's what we want.

**Reply by wimjongman** [reply]
Yes, I see.

**Reply by lcaron** [reply]
Hi Jochen

I'll have a look. Can you please tell me what is your OS and your Desktop Environment ?

Take care

**Reply by jochenberger** [reply]
Hi Laurent,

sure, I'm using Gnome 43 on Debian testing.

Cheers,
Jochen

**Reply by jochenberger** [reply]
That should have been Gnome 42, sorry.
